frontend/src/features/admin-form/responses/ResponsesPage/storage/utils/EncryptedResponseCsvGenerator/EncryptedResponseCsvGenerator.ts

```diff
@@ -112,7 +112,7 @@ export class EncryptedResponseCsvGenerator extends CsvGenerator {
     if (this.hasBeenProcessed) return
 
     // Create a header row in CSV using the fieldIdToQuestion map.
-    const headers = this.isMrf ? MRF_CSV_HEADERS : NON_MRF_CSV_HEADERS
+    const headers = this.isMrf ? [...MRF_CSV_HEADERS] : [...NON_MRF_CSV_HEADERS]
```

Comment:
note: the destructuring is necessary as we are mutating the array below. Destructuring creates a new array which prevents mutations from affecting the main const array. 

Ideally, we would avoid mutations in the code, but this is beyond the scope of this PR.

Reply:
Good explanation to highlight that we need to create new array as we are mutating it at a later stage. Would recommend to put a short note in the code.

Reply:
left a NOTE and TODO